### ISSUE TITLE
[poco] enabling MySQL support the right way

### ISSUE THREAD

**Issue by twMr7** [Microsoft/vcpkg#2084]
**Problem:**
    Current poco configuration can't find the vcpkg installed libmysql port. With the following patch, poco can successfully build with libmysql, but I am not so sure about this hard-coded "libmysql.lib" name approach. Is this the right way to patch poco?

```
diff --git a/ports/poco/portfile.cmake b/ports/poco/portfile.cmake
index 3ef0af92..7f4df990 100644
--- a/ports/poco/portfile.cmake
+++ b/ports/poco/portfile.cmake
@@ -22,6 +22,13 @@ else()
     set(POCO_STATIC OFF)
 endif()
 
+file(GLOB MYSQL_INCLUDE_DIR "${CURRENT_INSTALLED_DIR}/include/mysql")
+if(EXISTS ${MYSQL_INCLUDE_DIR})
+    # enabling MySQL support
+    set(MYSQL_LIB "${CURRENT_INSTALLED_DIR}/lib/libmysql.lib")
+    set(MYSQL_LIB_DEBUG "${CURRENT_INSTALLED_DIR}/debug/lib/libmysql.lib")
+endif()
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
@@ -30,6 +37,11 @@ vcpkg_configure_cmake(
         -DENABLE_SEVENZIP=ON
         -DENABLE_TESTS=OFF
         -DPOCO_UNBUNDLED=ON # OFF means: using internal copy of sqlite, libz, pcre, expat, ...
+        -DMYSQL_INCLUDE_DIR=${MYSQL_INCLUDE_DIR}
+    OPTIONS_RELEASE
+        -DMYSQL_LIB=${MYSQL_LIB}
+    OPTIONS_DEBUG
+        -DMYSQL_LIB=${MYSQL_LIB_DEBUG}
 )
 
 vcpkg_install_cmake()
```

**Comment by twMr7** [comment]
#2088

**Comment by twMr7** [comment]
merged in [4ff90ec](https://github.com/Microsoft/vcpkg/commit/4ff90ec4309df1d5937d62c52683e1147066445a)
